### ISSUE TITLE
Checking crates which warn about building in release mode now fails

### ISSUE THREAD

**Issue by mattico** [rust-lang/rust-analyzer#8516]
After updating to the latest RA I started getting this issue:

![image](https://user-images.githubusercontent.com/853158/114743689-87e78000-9d12-11eb-87be-a0795a4533e7.png)

It's a bit unusual but [synopsys-usb-otg prints a warning](https://github.com/stm32-rs/synopsys-usb-otg/blob/1ffb3c6df75702bf2f2c86224caec9c6c6abf69b/build.rs#L5-L7) if you're compiling in debug mode. RA interprets this as an error so I get an ⚠ in the status bar and a notification every time it runs. Completion and problem detection seem to still work, but it's annoying to get spurious error notifications.

Adding

    "rust-analyzer.checkOnSave.extraArgs": ["--release"], // synopsys-usb-otg requires release

doesn't seem to help. The check-on-save system doesn't error anyway, just the initial check when launching code or when clicking on the ⚠ rust-analyzer status bar segment to reload it.

It's a bit silly but an option to run cargo check with `--release` would help. Not sure what else to do.

**Comment by mattico** [comment]
Hold on, I completely missed the `error: could not compile lazy_static` in the notification, let me check the logs.

**Comment by mattico** [comment]
Yes, the notification is caused just by the synopsys-usb-otg warning:

```
[ERROR rust_analyzer::reload] failed to switch build data: cargo check failed:
warning: synopsys-usb-otg is being compiled in debug mode. This driver works reliably only in release mode!
    Checking mainboardfw v0.2.1 (C:\Users\matth\Code\mainboard-firmware\firmware)
error: could not compile `mainboardfw`

To learn more, run the command again with --verbose.
```

The lazy_static error was just from having multiple directories with different targets open in one vscode workspace (I think).

**Comment by flodiebold** [comment]
Can you try the current nightly?

**Comment by mattico** [comment]
With current nightly the rust-analyzer load status never stops spinning but the cargo check error doesn't appear.

**Comment by ChayimFriedman2** [comment]
Probably irrelevant by now.
